src/MUSI6106Exec/MUSI6106Exec.cpp

```diff
@@ -16,33 +19,61 @@ void    showClInfo ();
 // main function
 int main(int argc, char* argv[])
 {
-    std::string             sInputFilePath,                 //!< file paths
-                            sOutputFilePath;
+    std::string sInputFilePath,                 //!< file paths
+                sOutputFilePath;
 
-    static const int        kBlockSize = 1024;
+    static const int kBlockSize = 1024;
 
-    clock_t                 time = 0;
+    clock_t time = 0;
 
-    float                   **ppfAudioData = 0;
+    float **ppfAudioData = 0;
+    float **ppfFilterData = 0;
 
-    CAudioFileIf            *phAudioFile = 0;
-    std::fstream            hOutputFile;
+    CAudioFileIf *phAudioFile = 0;
+    CCombFilter *phFilterFile = 0;
```

Comment:
Adding to my earlier comment, I see you are instantiating the base class directly, and not using the interface class as I believe was required.

Reply:
Yes, it's true, I did manage to understand the instantiating process, thus ignored it.